addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/command/UdpCubeCommand.java

```diff
@@ -133,7 +135,7 @@ private void receiveUdpCommandResponse() {
                     } else {
                         // TODO: Further parsing of the other message types
                         commandResponse.put("messageResponse",
-                                Utils.getHex(message.substring(24).getBytes(StandardCharsets.UTF_8)));
+                                Utils.getHex(message.substring(20).getBytes(StandardCharsets.UTF_8)));
```

Comment:
Suggest to test for the message length instead of just shorten it to 20.
That way we avoid the out of bound all of the time.

Reply:
It's not shortening actually, but cutting off the remaining bytes. The message _must_ be at least 20 bytes long, because at that point, the first 20 bytes have already been parsed. 
If there is no more data after byte 20, this substring function would just return an empty string and never fail with an exception.

Therefore, IMO, an additional sanity check does not make sense here.

Reply:
I mean using something like  message.length() instead of hardcoding it to only 20 chars.
or  (I don't remember anymore if there is much garbage) Math.max(24, message.length() )

Reply:
I'm not getting your point. If you check the code before the location where I've changed it, you will see that the message parsing is all with hard-coded offsets because that's just how the message is structured. Then, after having process the first 20 bytes, the substring just takes the rest which is 0..n chars long. Before the fix, it lost the first 4 bytes or grabbed into emptiness. Now it works. That's all.

Reply:
:flushed: sorry you are right... I had in mind it was limiting the length rather than expanding it..

Reply:
No problem. Instead of "cutting off the remaining bytes" I should better have written "retrieving the remaining bytes" in my first reply.